attack_pattern.go

```diff
@@ -18,30 +26,45 @@ func NewAttackPattern(name string) (ap AttackPattern, err error) {
 	return
 }
 
+// STIXObjectType returns the group of STIX objects it belongs to (domain, relationship, transport)
+func (ap *AttackPattern) STIXObjectType() string {
+	return domainObject
+}
+
+// UnmarshalJSON implements the encoding/json Unmarshaler interface (https://golang.org/pkg/encoding/json/#Unmarshaler).
+//
+// It will take JSON and deserialize to an Object.  This should not be called directly, but instead
+// json.Unmarshal(b []byte, v interface{}) should be used.
+func (ap *AttackPattern) UnmarshalJSON(b []byte) error {
```

Comment:
Since Object defines an unmarshaler and is emedded, unmarshal has to be called on each embedded object/fields so they're not ignored.
https://stackoverflow.com/questions/52294027/golang-structure-doesnt-unmarshal-properly-when-using-a-custom-unmarshal-for-a